src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCommon.java

```diff
@@ -0,0 +1,299 @@
+                }
+                else {
+                    columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, 50, scale, dateTimeFormatter));
+                }
+                break;
+
+            // Redirect SQLXML as LONGNVARCHAR
+            // SQLXML is not valid type in TDS
+            case java.sql.Types.SQLXML:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.LONGNVARCHAR, precision, scale, dateTimeFormatter));
+                break;
+
+            // Redirecting Float as Double based on data type mapping
+            // https://msdn.microsoft.com/en-us/library/ms378878%28v=sql.110%29.aspx
+            case java.sql.Types.FLOAT:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.DOUBLE, precision, scale, dateTimeFormatter));
+                break;
+
+            // redirecting BOOLEAN as BIT
+            case java.sql.Types.BOOLEAN:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, java.sql.Types.BIT, precision, scale, dateTimeFormatter));
+                break;
+
+            default:
+                columnMetadata.put(positionInSource, new ColumnMetadata(colName, jdbcType, precision, scale, dateTimeFormatter));
+        }
+
+        loggerExternal.exiting(loggerClassName, "addColumnMetadata");
+    }
+
+    /**
+     * Set the format for reading in dates from the batch/file.
+     * 
+     * @param dateTimeFormat
+     *            format to parse data sent as java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+     */
+    public void setTimestampWithTimezoneFormat(String dateTimeFormat) {
+        DriverJDBCVersion.checkSupportsJDBC42();
+        loggerExternal.entering(loggerClassName, "setTimestampWithTimezoneFormat", dateTimeFormat);
+
+        this.dateTimeFormatter = DateTimeFormatter.ofPattern(dateTimeFormat);
+
+        loggerExternal.exiting(loggerClassName, "setTimestampWithTimezoneFormat");
+    }
+
+    /**
+     * Set the format for reading in dates from the batch/file.
+     * 
+     * @param dateTimeFormatter
+     *            format to parse data sent as java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+     */
+    public void setTimestampWithTimezoneFormat(DateTimeFormatter dateTimeFormatter) {
+        loggerExternal.entering(loggerClassName, "setTimestampWithTimezoneFormat", new Object[] {dateTimeFormatter});
+
+        this.dateTimeFormatter = dateTimeFormatter;
+
+        loggerExternal.exiting(loggerClassName, "setTimestampWithTimezoneFormat");
+    }
+
+    /**
+     * Set the format for reading in dates from the batch/file.
+     * 
+     * @param timeFormat
+     *            format to parse data sent as java.sql.Types.TIME_WITH_TIMEZONE
+     */
+    public void setTimeWithTimezoneFormat(String timeFormat) {
+        DriverJDBCVersion.checkSupportsJDBC42();
```

Comment:
No need for DriverJDBCVersion.checkSupportsJDBC42();

Reply:
done.